misk-hibernate/src/main/kotlin/misk/hibernate/Destination.kt

```diff
@@ -0,0 +1,61 @@
+package misk.hibernate
+
+data class Destination(
+  val keyspace: Keyspace?,
+  val shard: Shard?,
+  val tabletType: TabletType?
+) {
+  init {
+    if (shard != null) {
+      check(keyspace == shard.keyspace)
+    }
+  }
+
+  constructor(shard: Shard) : this(shard.keyspace, shard, null)
+
+  constructor(tabletType: TabletType) : this(null, null, tabletType)
+
+  override fun toString(): String {
+    val tabletType = if (tabletType != null) "@$tabletType" else ""
+    if (shard != null) {
+      return "$shard$tabletType"
+    }
+    if (keyspace != null) {
+      return "$shard$tabletType"
```

Comment:
nit: shard is null at this point. Did you mean "keyspace"?

Reply:
Good point. Lemme write some tests for the `toString` method too!

Reply:
there!